taf/repository_tool.py

```diff
@@ -676,13 +680,20 @@ def sign_role_yubikeys(
                           role's metadata
         - SigningError: If the number of signing keys is insufficient
         """
+        import taf.yubikey as yk
+
         role_obj = self._role_obj(role_name)
         threshold = self.get_role_threshold(role_name)
         if len(public_keys) < threshold:
             raise SigningError(
                 role_name,
                 f"Insufficient number of signing keys. Signing threshold is {threshold}.",
             )
+
+        if pins is not None:
+            for serial_num, pin in pins.items():
+                yk.add_key_pin(serial_num, pin)
+
         for index, public_key in enumerate(public_keys):
             if public_key is None:
                 from taf.yubikey import get_piv_public_key_tuf
```

Comment:
This import statement should be moved outside the for loop, perhaps to line 684?